src/styles/blocks/_main.scss

```
@@ -0,0 +1,52 @@
+@import "variables";
+@import "mixins";
+
+.calendar {
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  padding: 10px;
+  width: $side-size * 7 + $gap-for-day * 7;
+  position: relative;
+  left: 50%;
+  transform: translate(-50%,50%);
+  gap: $gap-for-day;
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    @include square($side-size);
+    background-color: #eee;
+    border: 1px solid black;
+    box-sizing: border-box;
+    transition: 0.5s;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        position: absolute;
+        content: "#{$i}";
+      }
+    }
+
+    &:hover {
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+  }
+
+  @each $day, $langth in $first-day {
+    &--start-day--#{$day} .calendar__day:first-child {
```

don't increase specificity if it's unnecessary
```suggestion
    &--start-day--#{$day} :first-child {
```

---

I see. Corrected.